app/services/concerns/xccdf/rule_references_rules.rb

```diff
@@ -9,7 +9,8 @@ module RuleReferencesRules
       def save_rule_references_rules
         @rule_references_rules ||= @op_rules.flat_map do |op_rule|
           rule_references_for(op_rule: op_rule).map do |reference|
-            ::RuleReferencesRule.new(rule_reference_id: reference.id,
+            ::RuleReferencesRule
+              .find_or_initialize_by(rule_reference_id: reference.id,
                                      rule_id: rule_for(ref_id: op_rule.id).id)
```

Comment:
This has the potential to really slow this method down, since it now executes N queries. If it becomes problematic, maybe we can introduce an `existing_rule_references_rule` method or something that uses `where` to find them all in one query.